zeppelin-web-angular/e2e/models/note-toc-page.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Locator, Page } from '@playwright/test';
+import { NotebookKeyboardPage } from './notebook-keyboard-page';
+
+export class NoteTocPage extends NotebookKeyboardPage {
+  readonly tocToggleButton: Locator;
+  readonly tocPanel: Locator;
+  readonly tocTitle: Locator;
+  readonly tocCloseButton: Locator;
+  readonly tocListArea: Locator;
+  readonly tocEmptyMessage: Locator;
+  readonly tocItems: Locator;
+  readonly codeEditor: Locator;
+  readonly runButton: Locator;
+  readonly addParagraphButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.tocToggleButton = page.locator('.sidebar-button').first();
```

Comment:
How about adding more specific label(e.g., `aria-label`) to the element and using that to select the element?